src/scranpy/analyze.py

```diff
@@ -0,0 +1,563 @@
+from concurrent.futures import ProcessPoolExecutor
+from dataclasses import dataclass
+from functools import singledispatch
+from typing import Any, Literal, Mapping, Optional, Sequence, Union
+
+import numpy as np
+from biocframe import BiocFrame
+from mattress import tatamize
+from singlecellexperiment import SingleCellExperiment
+
+from . import clustering as clust
+from . import dimensionality_reduction as dimred
+from . import feature_selection as feat
+from . import marker_detection as mark
+from . import nearest_neighbors as nn
+from . import normalization as norm
+from . import quality_control as qc
+from .types import is_matrix_expected_type
+
+__author__ = "ltla, jkanche"
+__copyright__ = "ltla"
+__license__ = "MIT"
+
+
+def run_tsne(obj, opts):
+    res = dimred.run_tsne(obj, opts)
+    return {"x": res.x, "y": res.y}
+
+
+def run_umap(obj, opts):
+    res = dimred.run_umap(obj, opts)
+    return {"x": res.x, "y": res.y}
+
+
+@dataclass
+class QualityControlOpts:
```

Comment:
Why are these at the top level, and not inside each subdirectory's `analyze.py`?